mittab/apps/tab/outround_pairing_views.py

```diff
@@ -0,0 +1,639 @@
+def pair_next_outround(request, num_teams, type_of_round):
+    if request.method == "POST":
+        backup.backup_round("before_pairing_%s_%s" %
+                            (num_teams / 2, type_of_round))
+
+        Outround.objects.filter(num_teams__lt=num_teams,
+                                type_of_round=type_of_round).delete()
+
+        outround_tab_logic.pair(type_of_round)
+
+        return redirect_and_flash_success(
+            request, "Success!", path=reverse("outround_pairing_view",
+                                              kwargs={
+                                                  "num_teams": int(num_teams / 2),
+                                                  "type_of_round": type_of_round
+                                              }))
+
+    # See if we can pair the round
+    title = "Pairing Outrounds"
+    current_round_number = 0
+
+    previous_round_number = TabSettings.get("tot_rounds", 5)
+
+    check_status = []
+
+    judges = outround_tab_logic.have_enough_judges_type(type_of_round)
+    rooms = outround_tab_logic.have_enough_rooms_type(type_of_round)
+
+    msg = "Enough judges checked in for Out-rounds? Need {0}, have {1}".format(
+        judges[1][1], judges[1][0])
+
+    if num_teams <= 2:
+        check_status.append(("Have more rounds?", "No", "Not enough teams"))
+    else:
+        check_status.append(("Have more rounds?", "Yes", "Have enough teams!"))
+
+    if judges[0]:
+        check_status.append((msg, "Yes", "Judges are checked in"))
+    else:
+        check_status.append((msg, "No", "Not enough judges"))
+
+    msg = "N/2 Rooms available Round Out-rounds? Need {0}, have {1}".format(
+        rooms[1][1], rooms[1][0])
+    if rooms[0]:
+        check_status.append((msg, "Yes", "Rooms are checked in"))
+    else:
+        check_status.append((msg, "No", "Not enough rooms"))
+
+    round_label = "[%s] Ro%s" % ("N" if type_of_round else "V",
+                                 num_teams)
+    msg = "All Rounds properly entered for Round %s" % (
+        round_label)
+    ready_to_pair = "Yes"
+    ready_to_pair_alt = "Checks passed!"
+    try:
+        outround_tab_logic.have_properly_entered_data(num_teams, type_of_round)
+        check_status.append((msg, "Yes", "All rounds look good"))
+    except PrevRoundNotEnteredError as e:
+        ready_to_pair = "No"
+        ready_to_pair_alt = str(e)
+        check_status.append(
+            (msg, "No", "Not all rounds are entered. %s" % str(e)))
+
+    return render(request, "pairing/pair_round.html", locals())
+
+
+@permission_required("tab.tab_settings.can_change", login_url="/403/")
+def break_teams(request):
+    if request.method == "POST":
+        # Perform the break
+        backup.backup_round("before_the_break")
+
+        teams, nov_teams = cache_logic.cache_fxn_key(
+            get_team_rankings,
+            "team_rankings",
+            request
+        )
+
+        nov_teams_to_break = TabSettings.get("nov_teams_to_break", 4)
```

Comment:
What do you think about just failing the request if these aren't specified? Since it changes so much from tournament-to-tournament I feel weird about having a default here

Reply:
Done